cypress/integration/4_findings.spec.js

```diff
@@ -0,0 +1,134 @@
+    // Click findingId to trigger Finding details flyout
+    cy.get(`[data-test-subj="finding-details-flyout-button"]`, { timeout: 2000 }).eq(0).click();
+
+    // Confirm flyout contents
+    cy.contains('Finding details');
+    cy.contains('Rule details');
+    cy.contains('Search Findings').should('not.exist');
+
+    // Close Flyout
+    cy.get(`[data-test-subj="close-finding-details-flyout"]`).then(($el) => {
+      cy.get($el).click({ force: true });
+    });
+
+    // wait for icon to become clickable - in this case, timeout insufficient.
+    cy.wait(1000);
+
+    // Click View details icon
+    cy.get(`[data-test-subj="view-details-icon"]`).eq(0).click({ force: true });
+
+    // Confirm flyout contents
+    cy.contains('Finding details');
+    cy.contains('Rule details');
+    cy.contains('Search Findings').should('not.exist');
+
+    // Close Flyout
+    cy.get(`[data-test-subj="close-finding-details-flyout"]`).then(($el) => {
+      cy.get($el).click({ force: true });
+    });
+  });
+
+  it('allows user to view details about rules that were triggered', () => {
+    // open Finding details flyout via finding id link. cy.wait essential, timeout insufficient.
+    cy.wait(1000);
+    cy.get(`[data-test-subj="view-details-icon"]`).eq(0).click({ force: true });
+
+    // open rule details inside flyout
+    cy.get('button', { timeout: 1000 });
+    // cy.get('.euiAccordion__button').contains('USB Device Plugged').click({ force: true });
+    cy.get(`[data-test-subj="finding-details-flyout-rule-accordion-0"]`).click({ force: true });
+
+    // Confirm content
+    cy.contains('Documents');
+    cy.contains('Detects plugged USB devices');
+    cy.contains('Low');
+    cy.contains('Windows');
+    cy.contains('cypress-test-windows');
+
+    ruleTags.forEach((tag) => {
+      cy.contains(tag);
+    });
+  });
+
+  // TODO - upon reaching rules page, trigger appropriate rules detail flyout
+  it('takes user to rules page when rule name inside accordion drop down is clicked', () => {
+    // Click rule link
+    cy.get(`[data-test-subj="finding-details-flyout-USB Device Plugged-details"]`)
+      .invoke('removeAttr', 'target')
+      .click({ force: true });
+
+    // Confirm destination reached
+    cy.url().should('include', 'opensearch_security_analytics_dashboards#/rules');
+  });
+
+  after(() => {
+    // Visit Detectors page
+    cy.visit(`${Cypress.env('opensearch_dashboards')}/app/${PLUGIN_NAME}#/detectors`, {
+      timeout: 5000,
+    });
+
+    cy.url().should('include', 'opensearch_security_analytics_dashboards#/detectors');
+
+    // Click on detector to be removed
+    cy.contains('sample_detector').click({ force: true }, { timeout: 2000 });
+
+    // Click "Actions" button, the click "Delete"
+    cy.get('button').contains('Actions').click({ force: true });
+    cy.contains('Delete').click({ force: true });
+
+    // Confirm detector deleted
+    cy.contains('There are no existing detectors.');
```

Comment:
This might not be true if any detectors are created while the tests are executing. It may be worthwhile to consider using the search bar on the detectors table to filter for the detector used for these tests, and ensure the table is empty.

Reply:
That sounds reasonable, in the interest of covering various cases.

Reply:
done